cmpserver/plugin/config.go

```diff
@@ -0,0 +1,86 @@
+package plugin
+
+import (
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/argoproj/argo-cd/v2/common"
+	configUtil "github.com/argoproj/argo-cd/v2/util/config"
+)
+
+const (
+	ConfigManagementPluginKind string = "ConfigManagementPlugin"
+)
+
+type PluginConfig struct {
+	metav1.TypeMeta `yaml:",inline"`
+	Metadata        metav1.ObjectMeta `yaml:"metadata"`
```

Comment:
Please use `json` tag instead for consistence with other data types. YAML marshaler works with json tag